src/core/model_standard.jl

```diff
@@ -154,7 +154,13 @@ function StandardABM(
     end
     !(is_sumtype(A)) && agent_validator(A, space, warn)
     C = construct_agent_container(container, A)
-    agents = C()
+    if C <: StructVector
+        fields_with_types = zip(fieldnames(A), fieldtypes(A))
+	tuple_init = NamedTuple(x => T[] for (x, T) in fields_with_types)
+	agents = C(tuple_init)
+    else
+        agents = C()
+    end
     agents_types = union_types(A)
     T = typeof(agents_types)
     return StandardABM{S,A,C,T,G,K,F,P,R}(agents, agent_step!, model_step!, space, scheduler,
```

Comment:
```suggestion
    return StandardABM{S,A,typeof(agents),T,G,K,F,P,R}(agents, agent_step!, model_step!, space, scheduler,
```

otherwise the type is abstract